internal/rpc/starknet/starknet.go

```diff
@@ -65,161 +69,162 @@ func (s *StarkNetRpc) GetStateUpdate(blockId *BlockId) (any, error) {
 		number, _ := blockId.number()
 		return s.synchronizer.GetStateDiff(int64(number)), nil
 	default:
-		// TODO: manage unexpected type
-		return nil, nil
+		return nil, InvalidBlockId
 	}
 }
 
-func (s *StarkNetRpc) GetStorageAt(blockId *BlockId, address string, key string) (any, error) {
-	// Parsing Key param
+func (s *StarkNetRpc) GetStorageAt(address string, key string, blockId *BlockId) (any, error) {
 	if !isStorageKey(key) {
-		// TODO: the rpc spec does not specify what to do if the key is not a storage key
-		return nil, nil
+		return nil, InvalidStorageKey
 	}
 	keyF := new(felt.Felt).SetHex(key)
-	// Parsing Address param
 	if !isFelt(address) {
-		return nil, NewContractNotFound()
+		return nil, ContractNotFound
 	}
 	addressF := new(felt.Felt).SetHex(address)
-	// Searching for the block in the database
-	b, err := getBlockById(blockId, s.blockManager)
+	b, err := getBlockById(blockId, s.blockManager, s.logger)
 	if err != nil {
 		return nil, err
 	}
-	// Building the state of the block
 	_state := state.New(s.stateManager, b.NewRoot)
-	// Searching for the value of the key in the state
 	value, err := _state.GetSlot(addressF, keyF)
 	if err != nil {
 		if errors.Is(err, db.ErrNotFound) {
-			return nil, NewContractNotFound()
+			return nil, ContractNotFound
 		}
-		// TODO: manage unexpected error
+		s.logger.Errorw(err.Error(), "function", "GetStorageAt")
+		return nil, UnexpectedError
 	}
 	return value.Hex0x(), nil
 }
 
 func (s *StarkNetRpc) GetTransactionByHash(transactionHash string) (any, error) {
-	// Parsing TransactionHash param
 	if !isFelt(transactionHash) {
-		return nil, NewInvalidTxnHash()
+		return nil, InvalidTxnHash
 	}
 	txHash := new(felt.Felt).SetHex(transactionHash)
 	tx, err := s.txnManager.GetTransaction(txHash)
 	if err != nil {
 		if errors.Is(err, db.ErrNotFound) {
-			return nil, NewInvalidTxnHash()
+			return nil, InvalidTxnHash
 		}
-		// TODO: manage unexpected error
+		s.logger.Errorw(err.Error(), "function", "GetTransactionByHash")
+		return nil, UnexpectedError
 	}
 	return NewTxn(tx)
 }
 
-func (s *StarkNetRpc) GetTransactionByBlockIdAndIndex(blockId *BlockId, index uint64) (any, error) {
-	b, err := getBlockById(blockId, s.blockManager)
+func (s *StarkNetRpc) GetTransactionByBlockIdAndIndex(blockId *BlockId, index *uint64) (any, error) {
```

Comment:
Why are we using a pointer to a `uint64`?

Reply:
If the user doesn't pass the `index` param, then the 0 value is set by default (which is valid). But `index` is a required param, so if the user doesn't pass it we need to return the `InvalidTxnIndex` error. Basically is to manage the edge case for 0 value.

Reply:
I see 😵.